includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1128,11 +1088,18 @@ public function ajax_add_to_cart() {
 				$data_store   = WC_Data_Store::load( 'product' );
 				$variation_id = $data_store->find_matching_product_variation( $product, $attributes );
 
+				if ( ! empty( $variation_id ) ) {
+					$variation        = wc_get_product( $variation_id );
+					$has_enough_stock = $variation->has_enough_stock( $qty );
+				}
 				WC()->cart->add_to_cart( $product->get_id(), $qty, $variation_id, $attributes );
+			} elseif ( 'simple' === $product_type || 'subscription' === $product_type ) {
+				WC()->cart->add_to_cart( $product->get_id(), $qty );
 			}
 
-			if ( 'simple' === $product_type || 'subscription' === $product_type ) {
-				WC()->cart->add_to_cart( $product->get_id(), $qty );
+			if ( ! $has_enough_stock ) {
+				/* translators: 1: product name 2: quantity in stock */
+				throw new Exception( sprintf( __( 'You cannot add that amount of "%1$s"; to the cart because there is not enough stock (%2$s remaining).', 'woocommerce-gateway-stripe' ), $product->get_name(), wc_format_stock_quantity_for_display( $product->get_stock_quantity(), $product ) ) );
```

Comment:
Shouldn't we throw this before attempting to add the product to the cart?

Reply:
Nice catch. I'll update the code